### PR TITLE
Fix #235: Enable global vlan

### LINKED ISSUE #235: Global VLAN
HI Guys,

We use a VLAN for mgmt which spans multiple sites, rather than adding the same VLAN and making it site specific which we would do 99% of the time, can we have an option in site to be global which would automatically apply across all sites, so we only have to add it once?

Thanks
Shaun

### PR DESCRIPTION
**Pull request by peng-shawn**
Decouple site/vlan, make site optional for vlan/vlangroup
Change html generation code to check site existence before
dereference
Create site search function, if site is None for a VLAN, view it as
global VLAN

### CODE REVIEW

**Comment by Shawn0720** [comment]
@drybjed, @hackman61. I created a new pull request for this issue. this one is based on the latest development branch. The rebase kinda messed up the other pull request timeline. This is my first pull request, still learning how to do it. Sorry for the confusion. 
@jeremystretch, it would be really helpful if you can provide some comment on this request. I will try to fix any issue in the diff. Or we can talk about design choice for this issue. Thanks for your time.

**Comment by jeremystretch** [comment]
Merging this into its own branch for evaluation.

File: netbox/ipam/models.py [hunk 1]
```diff
@@ -497,7 +497,8 @@ class Meta:
         verbose_name_plural = 'VLAN groups'
 
     def __str__(self):
-        return u'{} - {}'.format(self.site.name, self.name)
+        site_name = self.site.name if self.site else '__global'
```
Comment (jeremystretch) [inline netbox/ipam/models.py]: `__str__` should return something user-friendly. We can simply omit the site name for global VLANs.

File: netbox/templates/ipam/vlan.html [hunk 2]
```diff
@@ -8,7 +8,11 @@
...
         <ol class="breadcrumb">
             <li><a href="{% url 'ipam:vlan_list' %}">VLANs</a></li>
-            <li><a href="{% url 'ipam:vlan_list' %}?site={{ vlan.site.slug }}">{{ vlan.site }}</a></li>
+            {% if vlan.site %}
+                <li><a href="{% url 'ipam:vlan_list' %}?site={{ vlan.site.slug }}">{{ vlan.site }}</a></li>
+            {% else %}
+                <li><a href="{% url 'ipam:vlan_list' %}?site_id=0">Global</a></li>
+            {% endif %}
             {% if vlan.group %}
                 <li><a href="{% url 'ipam:vlan_list' %}?site={{ vlan.site.slug }}&group={{ vlan.group.slug }}">{{ vlan.group.name }}</a></li>
```
Comment (jeremystretch) [inline netbox/templates/ipam/vlan.html]: `vlan.site` could be None here

File: netbox/templates/ipam/vlan.html [hunk 3]
```diff
@@ -8,7 +8,11 @@
     <div class="col-sm-8 col-md-9">
         <ol class="breadcrumb">
             <li><a href="{% url 'ipam:vlan_list' %}">VLANs</a></li>
-            <li><a href="{% url 'ipam:vlan_list' %}?site={{ vlan.site.slug }}">{{ vlan.site }}</a></li>
+            {% if vlan.site %}
+                <li><a href="{% url 'ipam:vlan_list' %}?site={{ vlan.site.slug }}">{{ vlan.site }}</a></li>
+            {% else %}
+                <li><a href="{% url 'ipam:vlan_list' %}?site_id=0">Global</a></li>
```
Comment (jeremystretch) [inline netbox/templates/ipam/vlan.html]: Let's skip the "global" breadcrumb to be consistent with the way global prefixes are displayed

**Review by jeremystretch — COMMENTED** [review]
No need to resubmit the PR; I'm making the necessary changes in the branch. Just wanted to keep notes. Thanks for your contribution!

**Pull request merged**